backend/pkg/kafka/consumer.go

```diff
@@ -80,12 +85,22 @@ func (s *Service) FetchMessages(ctx context.Context, progress IListMessagesProgr
 	}
 	defer client.Close()
 
-	// 2. Setup JavaScript interpreter
-	isMessageOK, err := s.setupInterpreter(consumeRequest.FilterInterpreterCode)
-	if err != nil {
-		s.Logger.Error("failed to setup interpreter", zap.Error(err))
-		progress.OnError(fmt.Sprintf("failed to setup interpreter: %v", err.Error()))
-		return err
+	// 2. Create consumer workers
+	jobs := make(chan *kgo.Record, 100)
+	resultsCh := make(chan *TopicMessage, 100)
+	defer close(jobs)
```

Comment:
This can cause a panic in `consumeKafkaMessages`. The context is canceled immediately before jobs is closed; if goroutines are scheduled in a pathological enough order, then the select condition in `consumeKafkaMessages` may try sending to a closed channel.

It'd be easier to move the `defer close(jobs)` to the top of `consumeKafkaMessages`.